pkg/chunk/table_manager.go

```diff
@@ -197,6 +223,90 @@ func (m *TableManager) loop() {
 	}
 }
 
+func (m *TableManager) setRetentionMetrics() error {
```

Comment:
I think this is over complicated. I believe it goes a bit against what metrics monitoring should be: expose a metric gatherest in the most easieast and safest way. Here we do replicate some of the table manager to compute values (ie. `expectedMinIndexTableNumber`) and I'm not sure it's the right approach.

_I will reach you offline to discuss on what we want to really achieve with this PR, to offer some alternative way to do it._